src/grouped-legend-styles.ts

```diff
@@ -15,28 +15,47 @@ export class Styles {
     };
   }
 
+  get globalContainer(): Partial<CSSStyleDeclaration> {
+    const position = this.chartOptions.plugins?.groupedlegend?.position ?? 'top';
+    return {
+      display: 'flex',
+      flexDirection: ['top', 'bottom'].includes(position) ? 'column' : 'row',
+      flexGrow: '1',
+      flexShrink: '1',
+      overflow: 'hidden',
+    };
+  }
+
   get canvasContainer(): Partial<CSSStyleDeclaration> {
     return {
       maxHeight: '100%',
       minHeight: '0',
-      flexGrow: '1',
+      maxWidth: '100%',
+      minWidth: '0',
+      flex: '1',
     };
   }
 
   get legendContainer(): Partial<CSSStyleDeclaration> {
     const font = toFont(this.chartOptions.plugins?.legend?.labels?.font as Partial<FontSpec>);
+    const position = this.chartOptions.plugins?.groupedlegend?.position ?? 'top';
+    const display = this.chartOptions.plugins?.groupedlegend?.display ?? true;
     return {
-      display: 'flex',
+      display: display ? 'flex' : 'none',
       flexDirection: 'row',
-      alignItems: 'center',
-      justifyContent: 'center',
-      maxHeight: '100%',
+      flexWrap: 'wrap',
+      writingMode: ['top', 'bottom'].includes(position) ? 'horizontal-tb' : 'vertical-lr',
```

Comment:
That's a workaround for weird flexbox column & width behavior: https://stackoverflow.com/a/41209546/9375075